File: lib/src/main/java/uk/iatom/rfm69/registers/values/Listen2.java
package uk.iatom.rfm69.registers.values; import uk.iatom.rfm69.registers.IRegisterValue;

public enum Listen2 implements IRegisterValue {
    
    // RegListen2),
    COEFIDLE_VALUE(0xF5); // Default

    private byte val;
    
    Listen2(int val) {
        this.val = (byte) val;
    }

    @Override public byte val() {
        return val;
    }
}

File: lib/src/main/java/uk/iatom/rfm69/registers/values/Listen2.kt
package uk.iatom.rfm69.registers.values

import uk.iatom.rfm69.registers.IRegisterValue

enum class Listen2(`val`: Int): IRegisterValue { // RegListen2),
COEFIDLE_VALUE(0xF5);


    // Default
    private val `val`: Byte

    init {
        this.`val` = `val`.toByte()
    }

    override fun `val`(): Byte {
        return `val`
    }
}
